core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -169,15 +189,31 @@ private String getColSeparator(String format) {
     @Macro
     private String oracleServerUsername;
 
-    @Description("Password to use to connect to the remote Oracle Host via SSH.")
+    @Description("Mechanism to perform the secure shell action. Acceptable values are Private Key, Password.")
+    @Macro
+    private String authMechanism;
+
+    @Nullable
+    @Description("The password to be used to perform the secure shell action.")
     @Macro
     private String oracleServerPassword;
 
+    @Nullable
+    @Description("The private key to be used to perform the secure shell action.")
+    @Macro
+    private String privateKey;
+
+    @Nullable
+    @Description("Passphrase used to decrypt the provided private key. This should be kept blank when using Password " +
```

Comment:
Perhaps just say ".... This will be ignored when Password is used as the authentication mechanism." Shouldn't enforce that users should set it to empty, its unnecessary.

Applies to all similar fields.